components/builder-jobsrv/src/main.rs

```diff
@@ -36,13 +36,30 @@ fn main() {
     env_logger::init().unwrap();
     let matches = app().get_matches();
     debug!("CLI matches: {:?}", matches);
-    let config = match config_from_args(&matches) {
-        Ok(result) => result,
+    let (subcmd, config) = match subcmd_and_config_from_args(&matches) {
+        Ok((s, c)) => (s, c),
         Err(e) => return exit_with(e, 1),
     };
-    match jobsrv::server::run(config) {
-        Ok(_) => std::process::exit(0),
-        Err(e) => exit_with(e, 1),
+
+    match subcmd {
+        "migrate" => {
+            match jobsrv::server::migrate(config) {
+                Ok(_) => process::exit(0),
+                Err(e) => exit_with(e, 1),
+            }
+        }
+        "start" => {
+            match jobsrv::server::run(config) {
+                Ok(_) => process::exit(0),
+                Err(e) => exit_with(e, 1),
+            }
+        }
+        whoops => {
```

Comment:
Minor - should this be a _ instead or is it a catch-all?

Reply:
It is a catch-all, but I wanted to be able to refer to it in the subsequent `println!`, which is why I named it.

Reply:
And like I mentioned in the comment, it should never happen because clap will validate the subcommands, so if you enter `lol` for a subcommand, you get a clap error, but because Rust's compiler does exhaustiveness checking for `match` statements, you need some kind of catch-all here.

Reply:
👍

Reply:
That'll work, but you can also use the (albeit lazier) `unreachable!()` macro like we do in [`hab`](https://github.com/habitat-sh/habitat/blob/376054bdf3fe4ee36abaa7957dbe8962440078fb/components/hab/src/main.rs#L168). Granted it doesn't output a very nice result to the end user, but at that point we're into programmer/logic bug and not an expected user experience anyway.

Reply:
Oh nice!  I didn't know about `unreachable!()`.  That's perfect for this use case.